### ISSUE TITLE
Exceptions in Tweets2013 indexing

### ISSUE THREAD

**Issue by lintool** [castorini/Anserini#282]
Tweets2013 indexing gives a bunch of exceptions:

```
2018-06-28 11:37:47,773 INFO  [main] index.IndexCollection (IndexCollection.java:234) - Collection path: /tuna1/collections/twitter/Tweets2013-corpus/data/
2018-06-28 11:37:47,774 INFO  [main] index.IndexCollection (IndexCollection.java:235) - Index path: lucene-index.tweets2013.pos+docvectors+rawdocs
2018-06-28 11:37:47,775 INFO  [main] index.IndexCollection (IndexCollection.java:236) - CollectionClass: TweetCollection
2018-06-28 11:37:47,775 INFO  [main] index.IndexCollection (IndexCollection.java:237) - Generator: TweetGenerator
2018-06-28 11:37:47,776 INFO  [main] index.IndexCollection (IndexCollection.java:238) - Threads: 32
2018-06-28 11:37:47,776 INFO  [main] index.IndexCollection (IndexCollection.java:239) - Keep stopwords? false
2018-06-28 11:37:47,776 INFO  [main] index.IndexCollection (IndexCollection.java:240) - Store positions? true
2018-06-28 11:37:47,776 INFO  [main] index.IndexCollection (IndexCollection.java:241) - Store docvectors? true
2018-06-28 11:37:47,776 INFO  [main] index.IndexCollection (IndexCollection.java:242) - Store transformed docs? false
2018-06-28 11:37:47,777 INFO  [main] index.IndexCollection (IndexCollection.java:243) - Store raw docs? true
2018-06-28 11:37:47,777 INFO  [main] index.IndexCollection (IndexCollection.java:244) - Optimize (merge segments)? false
2018-06-28 11:37:47,777 INFO  [main] index.IndexCollection (IndexCollection.java:245) - Whitelist: null
2018-06-28 11:37:47,796 INFO  [main] index.IndexCollection (IndexCollection.java:276) - Starting indexer...
2018-06-28 11:37:48,469 INFO  [main] index.IndexCollection (IndexCollection.java:298) - 1416 files found in /tuna1/collections/twitter/Tweets2013-corpus/data
2018-06-28 11:37:55,257 WARN  [pool-2-thread-16] collection.Collection$FileSegment (Collection.java:101) - Exception when parsing document:null
2018-06-28 11:37:55,260 WARN  [pool-2-thread-16] collection.Collection$FileSegment (Collection.java:101) - Exception when parsing document:null
2018-06-28 11:37:55,273 WARN  [pool-2-thread-16] collection.Collection$FileSegment (Collection.java:101) - Exception when parsing document:null
...
```

We should look into what's going on... Tweet2011 doesn't have the same exceptions.

**Comment by borislin** [comment]
These are `NullPointerException` because the Tweet JSON data are not well structured, e.g. some data have missing fields as shown below:

![image](https://user-images.githubusercontent.com/767281/42358772-55644044-80ac-11e8-9e6a-26bb4270a0aa.png)

Can we all agree on what required fields should we have in order for a `TweetDocument` to be created?

**Comment by Peilin-Yang** [comment]
As per https://developer.twitter.com/en/docs/tweets/data-dictionary/overview/intro-to-tweet-json the following fields are fundamental fields.
```
{  
"created_at": "Thu May 10 15:24:15 +0000 2018",
 "id_str": "850006245121695744",
 "text": "Here is the Tweet message.",
 "user": {
    "id": 2244994945,
    "name": "Twitter Dev",
    "screen_name": "TwitterDev",
    "location": "Internet",
    "url": "https:\/\/dev.twitter.com\/",
 },
 "place": {
 },
 "entities": {
 },
 "extended_entities": {
 }
}
```
Where `created_at` should be a valid datetime string, `id_str` must not be empty, `user.id` must not be empty.

**Comment by borislin** [comment]
Fixed in #294
